testing/integration/ess/beat_receivers_test.go

```diff
@@ -911,6 +908,138 @@ agent.monitoring.enabled: false
 	assert.Equal(t, expectedMessage, message)
 }
 
+// TestComponentWorkDir verifies that the component working directory is not deleted when moving the component from
+// the process runtime to the otel runtime.
+func TestComponentWorkDir(t *testing.T) {
```

Comment:
This is very similar to what TestCoordinatorManagesComponentWorkDirs in the unit test covers isn't it? 

TestMonitoringNoDuplicates later on basically also requires this to work to pass? Is this test giving us anything that isn't already covered elsewhere?

Reply:
True. I wanted to have an integration test that explicitly tests this behavior for agent as a black box, but I can see the argument that it's redundant. If and when we decide to change this behavior, this test could also be expanded to cover that.